src/procedures/proc_functions.c

```
@@ -0,0 +1,152 @@
+	rax *functions = __aeRegisteredFuncs;
+	raxStart(&pdata->iter, functions);
+	raxSeek(&pdata->iter, "^", NULL, 0);
+	pdata->output = array_new(SIValue, 3);
+	_process_yield(pdata, yield);
+
+	ctx->privateData = pdata;
+	return PROCEDURE_OK;
+}
+
+
+// promote the rax iterator to the next function and return its name and mode.
```

cppcheck

[error] Memory leak

(memleak)